### PR TITLE
Realsense - Adding Realsense Depth and IRs cameras to CSD

### PR DESCRIPTION
**Pull request by otaviobp**
Moving realsense streams from sample to default CSD. Also doing some improvements and adding IR cameras.
@lucasdemarchi As you have asked me about that, can you take a look?

### CODE REVIEW

**Comment by fnoop** [comment]
Might also want to control color_enable_auto_exposure, the default settings for the rgb camera are unuseable outdoors.  I'm still struggling to get anything decent outdoors from the r200, but there's a good reference here to settings which are a start:
https://github.com/fnoop/vision_landing/blob/realsense/src/track_targets.cpp#L267

**Comment by fnoop** [comment]
Note you're hardcoding vaapih264enc which isn't available on all platforms - realsense can be (and is) used on platforms other than Intel.

**Comment by otaviobp** [comment]
Adding another commit using settings from configfile to create the gstreamer pipeline, so same settings used for v4l2 videos will be used in realsense depth and IRs cameras

**Comment by otaviobp** [comment]
As we are just exporting the IR and depth cameras I won't set the color auto exposure and auto white balance in that class. RGB camera is obtained using V4L2. Issue #58 was created for that.

File: samples/stream_realsense.cpp [hunk 1]
```diff
@@ -31,30 +32,63 @@
...
     for (int i = 0, end = WIDTH * HEIGHT; i < end; ++i) {
-        rgb[i].r = depth[i] * 60 / ONE_METER;
-        rgb[i].g /= 4;
-        rgb[i].b /= 4;
+        uint8_t rainbow[3];
+        rainbow_scale((double)depth[i] * 0.001, rainbow);
+
+        ctx->rgb_data[i].r = rainbow[0];
+        ctx->rgb_data[i].g = rainbow[1];
+        ctx->rgb_data[i].b = rainbow[2];
```
Comment (lucasdemarchi) [inline samples/stream_realsense.cpp]: this looks heavy and a good candidate for parallelization.... but ok

File: Makefile.am [hunk 2]
```diff
@@ -126,7 +126,17 @@ AM_LDFLAGS = \
...
+if HAVE_REALSENSE
+REALSENSE_SRC += \
+	src/stream_builder_realsense.h \
+	src/stream_builder_realsense.cpp \
+	src/stream_realsense.cpp \
+	src/stream_realsense.h
+endif
+
 BASE_FILES = \
+	$(REALSENSE_SRC) \
```
Comment (lucasdemarchi) [inline Makefile.am]: REALSENSE_SRC is only used here. would be simpler IMO to just do:

```
if HAVE_REALSENSE
BASE_FILES += \
	src/stream_builder_realsense.h \
	src/stream_builder_realsense.cpp \
	src/stream_realsense.cpp \
	src/stream_realsense.h
endif
```

**Comment by lucasdemarchi** [comment]
LGTM

File: samples/stream_realsense.cpp [hunk 3]
```diff
@@ -1,165 +0,0 @@
...
-    rs_device *dev = rs_get_device(ctx, 0, NULL);
-    if (!dev) {
-        log_error("Unable to access realsense device");
-        return nullptr;
-    }
-
-    /* Configure all streams to run at VGA resolution at 60 frames per second */
-    rs_enable_stream(dev, RS_STREAM_DEPTH, WIDTH, HEIGHT, RS_FORMAT_Z16, 60, NULL);
-    rs_enable_stream(dev, RS_STREAM_COLOR, WIDTH, HEIGHT, RS_FORMAT_RGB8, 60, NULL);
-    rs_start_device(dev, NULL);
```
Comment (fnoop) [inline samples/stream_realsense.cpp]: FPS should be configurable.  According to the recent realsense paper from Intel, the framerate has a big effect on the quality of depth data.

File: Makefile.am (hunk 2 above)
Comment (otaviobp) [inline Makefile.am]: true

**Comment by otaviobp** [comment]
merged. I'll address FPS, width and height update in a different PR

**Pull request closed without merging**